### PR TITLE
FEATURE : added filter and searchbar for task page

### PR DESCRIPTION
**Pull request by sahsisunny**
### Description:

In this PR we have added the filter and search(Only for status) functionality to the tasks page. The filter and search functionality is similar to the GitHub issue/pulls filter and search bar. The filter section allows users to select various filtering options such as status (available, completed, in progress, etc).

### Anything you would like to inform the reviewer about:
- This feature is currently behind the feature flag.

### Dev Tested:
- [x] Yes

#### Test Coverage
imageimage

### Images/video of the change:

https://github.com/Real-Dev-Squad/website-status/assets/70854507/9494dc63-746d-4f01-ab2d-1325d9a8eede

### CODE REVIEW

File: src/components/tasks/TasksContent.tsx [hunk 1]
```diff
@@ -15,14 +15,16 @@ import { getActiveTab, tabToUrlParams } from '@/utils/getActiveTab';
...
-export const TasksContent = () => {
+const getQueryParamValue = (tab: Tab) => `is:${tabToUrlParams(tab)}`;
+
+export const TasksContent = ({ dev }: { dev: boolean }) => {
     const router = useRouter();
-    const { section }: routerQueryParams = router.query;
-    const selectedTab = getActiveTab(section);
+    const allQueryParams = router.query;
+    const q = allQueryParams.q as string;
+    const queryTab = q?.replace('is:', '');
```
Comment (Pratiyushkumar) [inline src/components/tasks/TasksContent.tsx:26]: NIT - is it possible to rename `q` with some better naming?

**Review by Pratiyushkumar — COMMENTED** [review]
I have suggested one change you can choose to change or not changes as per your wish

**Review by Pratiyushkumar — DISMISSED** [review]
rest looks good to me

File: src/components/tasks/TasksContent.tsx (hunk 1 above)
Comment (Pratiyushkumar) [reply src/components/tasks/TasksContent.tsx:26]: maybe `q-value` or `q-param-value` or `queryParam`

Comment (sahsisunny) [reply src/components/tasks/TasksContent.tsx]: Done

**Review by Pratiyushkumar — DISMISSED** [review]

**Review by shubhamsigdar1 — DISMISSED** [review]

File: src/components/tasks/TaskSearch/TaskSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+import className from './tasksearch.module.scss';
+import { TABS, Tab } from '@/interfaces/task.type';
+import FilterModal from './FilterModal';
+
+type TaskSearchProps = {
+    onSelect: (tab: Tab) => void;
+    inputtedValue: string;
```
Comment (vinit717) [inline src/components/tasks/TaskSearch/TaskSearch.tsx:8]: can we just use inputValue

File: src/components/tasks/TaskSearch/TaskSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    activeTab?: Tab;
+    onInputChange: (value: string) => void;
+    onClickSearchButton: () => void;
+};
+
+const TaskSearch = ({
+    onSelect,
+    inputtedValue,
```
Comment (vinit717) [inline src/components/tasks/TaskSearch/TaskSearch.tsx:16]: +1

File: src/components/tasks/TaskSearch/TaskSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                        />
+                    )}
+                </div>
+
+                <input
+                    className={className['search-input']}
+                    data-testid="search-input"
+                    type="text"
+                    placeholder="Eg: is:active assignee:sunny-s key:task"
+                    value={inputtedValue}
```
Comment (vinit717) [inline src/components/tasks/TaskSearch/TaskSearch.tsx:54]: +1

File: src/components/tasks/TasksContent.tsx [hunk 5]
```diff
@@ -15,14 +15,16 @@ import { getActiveTab, tabToUrlParams } from '@/utils/getActiveTab';
...
+const getQueryParamValue = (tab: Tab) => `is:${tabToUrlParams(tab)}`;
+
+export const TasksContent = ({ dev }: { dev: boolean }) => {
     const router = useRouter();
-    const { section }: routerQueryParams = router.query;
-    const selectedTab = getActiveTab(section);
+    const allQueryParams = router.query;
+    const qQueryParam = allQueryParams.q as string;
+    const queryTab = qQueryParam?.replace('is:', '');
+    const selectedTab = getActiveTab(queryTab);
```
Comment (vinit717) [inline src/components/tasks/TasksContent.tsx:27]: Can you just create a helper function for that? for simplicity

File: src/components/tasks/TaskSearch/tasksearch.module.scss [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    color: $black-color;
+    font-weight: bold;
+    cursor: pointer;
+    border: none;
+    border-right: none;
+    outline: none;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    padding-left: 0.3rem;
```
Comment (vinit717) [inline src/components/tasks/TaskSearch/tasksearch.module.scss:44]: for small value can you use pixel itself

File: src/components/tasks/TaskSearch/tasksearch.module.scss [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+    &:after {
+        content: '';
+        border-left: 0.3rem solid transparent;
+        border-right: 0.3rem solid transparent;
+        border-top: 0.3rem solid $black-color;
+        margin-left: 0.5rem;
```
Comment (vinit717) [inline src/components/tasks/TaskSearch/tasksearch.module.scss:51]: +1

File: src/components/tasks/TasksContent.tsx (hunk 5 above)
Comment (vinit717) [reply src/components/tasks/TasksContent.tsx:27]: `export const extractTabFromQuery = (queryParams) => {
  const qQueryParam = queryParams.q as string;
  const queryTab = qQueryParam?.replace('is:', '');
  return getActiveTab(queryTab);
};`

**Review by vinit717 — APPROVED** [review]

**Review by shubhamsigdar1 — APPROVED** [review]

**Pull request merged**